### ISSUE TITLE
nginx module: ACME is not working for me

### ISSUE THREAD

**Issue by 21stce** [NixOS/nixpkgs#164770]
### Describe the bug
When I set 

```
services.nginx.virtualHosts."example.com".enableACME = true; 
```

in a configuration that otherwise works fine. 

I get:

```
/build/.attr-0: line 1: www.example.com-start: command not found
```

I have tried replacing `"example.com"` by `"example"`, but this had no effect.

### Notify maintainers

@mattchrist 

### Metadata
21.11

**Comment by Izorkin** [comment]
The error is not reproduced.

**Comment by 21stce** [comment]
How should I make this work if I run my own nameserver?

**Comment by Izorkin** [comment]
Can you show the config?

**Comment by 21stce** [comment]
It's likely that there is nothing wrong with the module. I will have to check some more things first.

**Comment by ajs124** [comment]
This sounds like a problem with your setup. SERVFAIL sounds like your DNS Server does not accept the RFC2136 DNSUPDATE. Is there maybe an issue with your TSIG key or some ACL or something?

In any case, this is most likely not an issue in the acme or nginx module.

**Comment by 21stce** [comment]
The issue is in the NixOS module for bind being broken. (I opened an issue a week ago, but I suspect that the "maintainer" doesn't actually "maintain".) In order to fix it, the whole module's architecture and assumptions need to be changed, which brings me to the question whether there is any DNS server that *does* have a NixOS module that is actually ready for production use.

**Comment by ajs124** [comment]
> I suspect that the "maintainer" doesn't actually "maintain"

Yeah, that's kind of problem in NixOS/nixpkgs. Our "maintainer model" isn't too well-defined.

I haven't taken a look at the bind module, but we (@helsinki-systems) use knot-dns for authoritative, which is quite well maintained in nixpkgs, because the nixpkgs maintainer is an upstream maintainer. We used to use knot-resolver for recursing, but since that had an upstream bug, we switched to unbound and pdns-recursor.

**Comment by iblech** [comment]
> I suspect that the "maintainer" doesn't actually "maintain"

I feel your frustration and am sorry the module doesn't work as expected. NixOS and its ecosystem is always exclusively developed by people who have other day jobs or commitments and can only work on NixOS in their free time. The tone of your comment makes it seem like the rest of our ecosystem gave you the impression that most of us are paid full-time to work on NixOS, which I appreciate. :-)

Unfortunately, like in any collaborative open source endeavor, there are just the following options to get issues resolved: hope that others find the free time and energy to work on a fix; pay the current maintainers; or pay other people to help. On the bright side, most of the time the process works quite smoothly, many people can rely on NixOS for their daily driver, and improvements and new community members are always welcomed. :-)

Regarding the concrete issue at hand, I hope that it will get resolved soon. I have never worked with bind hence don't feel comfortable touching the bind module.

**Comment by 21stce** [comment]
@iblech There is absolutely no requirement for anyone to say they maintain anything, *but* the moment you do this, it creates expectations *and* it opens you up to criticism. That's how it works. So, if a maintainer does something useful, they might get praise, but if they do a bad job (and the bind module *is* a bad job), then they also get to hear that. A maintainer that doesn't maintain, should just be removed. If someone at a homeless shelter starts to abuse homeless people, they also get kicked out. Open-source is no different.  

Why would *anyone* want to pay the current maintainers (of the bind module) to do anything? Payment would be in order if they had some highly proprietary system and saw that the open-source module was managed perfectly by someone. I will just use whatever works and I want to thank @ajs124 for telling me what works for them.

**Comment by 21stce** [comment]
@AleXoundOS If you have an opinion, feel free to share it, but don't use emoticons like you are a child and have difficulty expressing yourself.

**Comment by 7c6f434c** [comment]
Looking through the issues you have opened, the BIND module is perfectly fine for its initially intended scope under its initially intended assumptions. There is very definitely no expectation here in Nix* community that maintainers implement clear feature requests via breaking changes. Telling all they know to someone trying to investigate a bug within the initial scope. reviewing the proposed patches that fix bugs or extend the featureset without breaking — that's one thing, and generally expected. What you expect is a different thing and the project will break down if we try to expect it from maintainers without creating more reasonable tiers of involvement.

**Comment by 21stce** [comment]
@7c6f434c The only correct "initial scope" for a module is one in which there is full support for every feature over time, IMO. The bind module can't support that in its current form. I don't understand how anyone could ever have a different opinion regarding this. We want to have *working* software, right? We don't want to be known "as that distribution with half-assed modules", right?

**Comment by 7c6f434c** [comment]
>The only correct "initial scope" for a module is one in which there is full support for every feature over time, IMO.

No. At some level of complexity, having an abstraction that restricts what is done but covers the frequent usecases is better. It's not like it is hard to write a module if you already have your preferred config (for your system, not something tunable enough for everyone) and starting command-line arguments, after all.

> I don't understand how anyone could ever have a different opinion regarding this.

If you find yourself having this thought on _any_ technical topic, you are going to be surprised a lot here…

> We don't want to be known "as that distribution with half-assed modules", right?

Actually that would be good. That distribution with half-assed and restricted implementation of what's quarter-assed elsewhere. There are people for whom with all the trade-offs NixOS is still worth it, and people for whom the trade-offs made here are unfitting.

> Using emoticons is a personal attack.

In this community, use of emoticons is interpreted to express an opinion about a statement, and it is considered normal. Disagreement with a statement is not equivalent to personal attacks. Yes, sure, you can find communities with other norms. As for most of communication.

> Perhaps it should be made clear what this "responsibility" is then, because the word "maintainer" has a pre-existing semantics in literally every software project in the world (as well as in packages like gdb maintained (indeed, maintained) by Red Hat), which is consistent with my usage of the word.

There are other opensource projects with «will review» concept of maintainership in practice. Yeah, probably could be nice to have something written down in the manual.

**Comment by 21stce** [comment]
Using an emoticon without any message only encourages group anger; it's damaging. 

Look at what you did instead: You can have a different idea about things and we have even come up with a way to resolve this minor issue. If you had just written down an emoticon, you would just  have created a reason to hate you and literally nothing else.

**Comment by 7c6f434c** [comment]
The main point of your message was to state that in your opinion scope-restriction of bind module is approaching abuse, and to wonder whether anyone would interpret the situation differently. Yes, expression of disagreement without further details _is_ information in such context, and if you refuse to extract it, and only find it grounds for personal hate, that's a pity.

**Comment by 21stce** [comment]
No, that was not the main point of my message, because I wasn't wondering. I was *stating* that my beliefs are such that maintainers should maintain or get out. You said that "reviewing", which is a subset of the tasks of a traditional maintainer is equivalent to maintaining in some projects, including quite possibly nixpkgs. 

I think it's not smart to redefine words, but if one does so without saying that, one is just trolling. The "you" is whoever downgraded the concept of "maintainer" without clearly defining that; if I were to ever "maintain" a package in this manner, I'd add a comment to with a link to this discussion. That way at least the expectations should be clear for everyone.  

Thank you for clarifying your position and I suppose next time I should first figure out whether for a given package someone is a maintainer or a reviewer.
